### ISSUE TITLE
Support Pathlike objects and names with .qu suffix in qsave/qload

### ISSUE THREAD

**Issue by labay11** [qutip/qutip#1811]
### Problem Description

Right now, if you pass a [pathlib.Path](https://docs.python.org/3/library/pathlib.html) to any of two functions above, an error will raise because it is not possible to add the suffix `.qu` to a Path object.

Moreover, if you already have the fullpath of the file (including the extension), QuTip will also raise because it appends the extension internally, no matter if it is already there or not.

### Proposed Solution

The solution would be similar to [numpy.save](https://github.com/numpy/numpy/blob/v1.22.0/numpy/lib/npyio.py#L449) method.

```
if file is string:
    if not file.endswith('.qu'):
          file = file + '.qu'
   file = Path(file)

-> open path object
```

### Additional Context

Pd: I'll be happy to make my first contribution and create a pull request for this.

**Comment by hodgestar** [comment]
A pull request would be most welcome. There are also two other functions in qutip.fileio that should probably be updated at the same time. Instead of using an if statement, I would write code like:
```python
p = pathlib.Path("filename")
...  # append suffix here
with p.open(...):
    ... # load or save here
```

**Comment by labay11** [comment]
Nice, I will do it in the following days.

On a side note, I'd propose to make the printing optional in `qload` with a `debug` flag for instance. As it is a bit messy when you are loading a lot of objects.

**Comment by hodgestar** [comment]
+1 on getting rid of the print statements. I think we can just drop them entirely -- users can just print their object if they want to see what was loaded.

**Comment by hodgestar** [comment]
Fixed #1813 and released in QuTiP 4.7.
